### PR TITLE
feat: `chameleon`, a tool for language binding generation

### PR DESCRIPTION
**Pull request by c-dilks**
This introduces `chameleon`, a tool to automate generation of bindings and action function tests. For a proof of concept, this PR generates the C bindings for Fortran.

### CODE REVIEW

**Review by rtysonCLAS12 — CHANGES_REQUESTED** [review]
Sorry I took so long to review, I forgot about this!
I checked the Action.yaml files, they have the right entries for each action function.

Should the ExampleAlgorithm have an Action.yaml file? This could point to the [action function Filter ](https://github.com/JeffersonLab/iguana/blob/6a7d113c7150b1251a1d0d0fdd4c438033f4165e/src/iguana/algorithms/example/ExampleAlgorithm/Algorithm.cc#L107), I think it would be helpful to refer to when writing algorithms.

**Comment by c-dilks** [comment]
> Should the ExampleAlgorithm have an Action.yaml file? This could point to the [action function Filter ](https://github.com/JeffersonLab/iguana/blob/6a7d113c7150b1251a1d0d0fdd4c438033f4165e/src/iguana/algorithms/example/ExampleAlgorithm/Algorithm.cc#L107), I think it would be helpful to refer to when writing algorithms.

Yes, but I'm actually considering removing `ExampleAlgorithm` with #141, in favor of choosing three representative algorithms (a filter, a transformer, and a creator) that would serve as good examples, and moving the descriptive comments to those. Then I'll add links to those in the documentation.

**Review by rtysonCLAS12 — APPROVED** [review]
OK sounds good

File: src/iguana/algorithms/clas12/LorentzTransformer/Algorithm.h [hunk 1]
```diff
@@ -31,19 +31,19 @@ namespace iguana::clas12 {
...
       void Stop() override;
 
-      /// @action_function{scalar transformer} boost the 4-momentum @f$p=(p_x,p_y,p_z,E)@f$ along @f$\beta=(\beta_x, \beta_y, \beta_z)@f$
-      /// @param p_x @f$p_x@f$
-      /// @param p_y @f$p_y@f$
-      /// @param p_z @f$p_z@f$
+      /// @action_function{scalar transformer} boost the 4-momentum @f$p=(px,py,pz,E)@f$ along @f$\beta=(\beta_x, \beta_y, \beta_z)@f$
+      /// @param px @f$px@f$
+      /// @param py @f$py@f$
+      /// @param pz @f$pz@f$
```
Comment (c-dilks) [inline src/iguana/algorithms/clas12/LorentzTransformer/Algorithm.h:37]:
```suggestion
      /// @action_function{scalar transformer} boost the 4-momentum @f$p=(p_x,p_y,p_z,E)@f$ along @f$\beta=(\beta_x, \beta_y, \beta_z)@f$
      /// @param px @f$p_x@f$
      /// @param py @f$p_y@f$
      /// @param pz @f$p_z@f$
```

**Pull request merged**